packages/core-data/src/use-entity-mutation.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Internal dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as coreStore } from '../';
+
+export default function useEntityMutation( kind, type, id ) {
+	const mutationState = useSelect(
+		( select ) => {
+			const args = [kind, type, id];
+			return {
+				editedRecord: select( coreStore ).getEditedEntityRecord( ...args ),
+				hasEdits: select( coreStore ).hasEditsForEntityRecord( ...args ),
+				isSaving: select( coreStore ).isSavingEntityRecord( ...args ),
+				saveError: select( coreStore ).getLastEntitySaveError( ...args ),
+				isDeleting: select( coreStore ).isDeletingEntityRecord( ...args ),
+				deleteError: select( coreStore ).getLastEntityDeleteError( ...args ),
```

Comment:
I always wonder why we need so many selectors 🤔 . Could it just be something like...

```typescript
{
  status: 'IDLE' | 'LOADING' | 'ERROR' | 'SUCCESS';
  error?: Error;
  data?: Response;
}
```

We can still provide `isLoading`, `isSuccess`, `isError`, etc for convenience just like in `react-query`.